File: generator/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        return sum(
            1
            for j in (
                row.index(1)
                for row in mat
                if row.count(1) == 1
            )
            if sum(
                1
                for i in range(len(mat))
                if mat[i][j]
            ) == 1
        )

File: ranges/1582-special-positions-in-a-binary-matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat)
    {
        return ranges::distance(
            mat
            | views::filter([](span<int> row) { return ranges::count(row, 1) == 1; })
            | views::transform([](span<int> row) { return ranges::find(row, 1) - row.begin(); })
            | views::filter(
                [&mat](auto j) {
                    return ranges::count(
                        mat | views::transform([j](span<int> row) { return row[j]; } ),
                        1
                    ) == 1;
                }
            )
        );
    }
};
